Assets/BossRoom/Scripts/Client/UI/Lobby/CreateLobbyUI.cs

```diff
@@ -0,0 +1,188 @@
+
+
+        /// <summary>
+        /// Sanitize user port InputField box allowing only alphanumerics, plus any matching chars, if provided.
+        /// </summary>
+        /// <param name="dirtyString"> string to sanitize. </param>
+        /// <param name="includeChars"> Array of chars to include. </param>
+        /// <returns> Sanitized text string. </returns>
+        private static string Sanitize(string dirtyString, char[] includeChars = null)
+        {
+            var result = new StringBuilder(dirtyString.Length);
+            foreach (char c in dirtyString)
+            {
+                if (char.IsLetterOrDigit(c) ||
+                    (includeChars != null && Array.Exists(includeChars, includeChar => includeChar == c)))
+                {
+                    result.Append(c);
+                }
+            }
+
+            return result.ToString();
+        }
+
+        /// <summary>
+        /// Added to the InputField component's OnValueChanged callback for the Room/IP UI text.
+        /// </summary>
+        public void SanitizeInputText()
+        {
+            var inputFieldText = Sanitize(m_IPInputField.text, k_InputFieldIncludeChars);
+            m_IPInputField.text = inputFieldText;
+        }
+
+        /// <summary>
+        /// Added to the InputField component's OnValueChanged callback for the Port UI text.
+        /// </summary>
+        public void SanitizePortText()
+        {
+            var inputFieldText = Sanitize(m_PortInputField.text);
+            m_PortInputField.text = inputFieldText;
+        }
+
+        #endregion
+
+        [Inject]
+        private void InjectDependencies(LobbyUIManager lobbyUIManager)
+        {
+            m_LobbyUIManager = lobbyUIManager;
+        }
+
+        private void Awake()
+        {
+            SetOnlineMode(OnlineMode.UnityRelay);
+            m_IPToggle.onValueChanged.AddListener(IPRadioRadioButtonPressed);
+            m_UnityRelayToggle.onValueChanged.AddListener(UnityRelayRadioRadioButtonPressed);
+        }
+
+        private void IPRadioRadioButtonPressed(bool value)
+        {
+            if (!value)
+            {
+                return;
+            }
+
+            SetOnlineMode(OnlineMode.IpHost);
+        }
+
+        private void UnityRelayRadioRadioButtonPressed(bool value)
+        {
+            if (!value)
+            {
+                return;
+            }
+
+            SetOnlineMode(OnlineMode.UnityRelay);
+        }
+
+
+        private void SetOnlineMode(OnlineMode mode)
+        {
+            m_OnlineMode = mode;
+
+            switch (mode)
+            {
+                case OnlineMode.IpHost:
+                {
+                    m_PortInputField.text = k_DefaultPort.ToString();
+                    m_IPInputField.gameObject.SetActive(true);
+
+                    m_PortInputField.text = k_DefaultPort.ToString();
+                    m_PortInputField.gameObject.SetActive(true);
+                }
+                    break;
```

Comment:
This logic for enabling and disabling fields is specific to the OnlineMode.  In the `UnityRelay` case, it's actually logic to disable the `IpHost` properties.  

Could we consider moving the logic directly into the `BlahRadioButtonPressed` functions (or a similarly named helper) and then you can include both parts.  E.g. If `value` is true, enable the bits, if it's `false disable the bits.  That way if there's unique UI for either one you don't need to mix the logic about which pieces to enable/disable.